modules/raft/src/main/java/org/apache/ignite/raft/jraft/rpc/impl/RaftGroupServiceImpl.java

```diff
@@ -613,7 +613,7 @@ else if (th instanceof SMFullThrowable)
      * @return {@code True} if this is a recoverable exception.
      */
     private boolean recoverable(Throwable t) {
-        return t.getCause() instanceof IOException || t.getCause() instanceof TimeoutException;
+        return t instanceof TimeoutException || (t.getCause() != null && t.getCause() instanceof IOException);
```

Comment:
You don't need to check `t.getCause()` for `null` value before using `instanceof`. So, this expression can be simplified:
`return t instanceof TimeoutException || t.getCause() instanceof IOException;`

Reply:
Fixed.